larpandoracontent/LArTwoDReco/LArCosmicRay/ClusterAssociation.h

```diff
@@ -0,0 +1,285 @@
+
+/**
+ *  @brief  ClusterPairAssociation class
+ */       
+class ClusterPairAssociation : public ClusterAssociation
+{
+public:
+    /**
+     *  @brief  Default constructor
+     */
+    ClusterPairAssociation();
+
+    /**
+     *  @brief  Constructor
+     *
+     *  @param  upstreamMergePoint the upstream merge point
+     *  @param  upstreamMergeDirection the cluster direction at the upstream merge point
+     *  @param  downstreamMergePoint the downstream merge point
+     *  @param  downstreamMergeDirection the cluster direction at the downstream merge point
+     *  @param  pUpstreamCluster the address of the upstream cluster
+     *  @param  pDownstreamCluster the address of the downstream cluster
+     */    
+    ClusterPairAssociation(const pandora::CartesianVector &upstreamMergePoint, const pandora::CartesianVector &upstreamMergeDirection,
+        const pandora::CartesianVector &downstreamMergePoint, const pandora::CartesianVector &downstreamMergeDirection, const pandora::Cluster *pUpstreamCluster,
+        const pandora::Cluster *pDownstreamCluster);
+
+    /**
+     *  @brief  Returns the address of the upstream cluster
+     *
+     *  @return  the address of the upstream cluster
+     */
+    const pandora::Cluster *GetUpstreamCluster() const;
+
+    /**
+     *  @brief  Returns the address of the downstream cluster
+     *
+     *  @return  the address of the downstream cluster
+     */
+    const pandora::Cluster *GetDownstreamCluster() const;    
+
+private:
+    const pandora::Cluster    *m_pUpstreamCluster;      ///< The address of the upstream cluster
+    const pandora::Cluster    *m_pDownstreamCluster;    ///< The address of the downstream cluster
+};
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+    
+inline ClusterAssociation::ClusterAssociation() :
+    m_upstreamMergePoint(pandora::CartesianVector(0.f, 0.f, 0.f)),
+    m_upstreamMergeDirection(pandora::CartesianVector(0.f, 0.f, 0.f)),
+    m_downstreamMergePoint(pandora::CartesianVector(0.f, 0.f, 0.f)),
+    m_downstreamMergeDirection(pandora::CartesianVector(0.f, 0.f, 0.f)),
+    m_connectingLineDirection(pandora::CartesianVector(0.f, 0.f, 0.f))
+{
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+inline ClusterAssociation::ClusterAssociation(const pandora::CartesianVector &upstreamMergePoint, const pandora::CartesianVector &upstreamMergeDirection,
+        const pandora::CartesianVector &downstreamMergePoint, const pandora::CartesianVector &downstreamMergeDirection) :
+    m_upstreamMergePoint(upstreamMergePoint),
+    m_upstreamMergeDirection(upstreamMergeDirection),
+    m_downstreamMergePoint(downstreamMergePoint),
+    m_downstreamMergeDirection(downstreamMergeDirection),
+    m_connectingLineDirection(0.f, 0.f, 0.f)
```

Comment:
Given that this is overwritten in the body of the constructor, no need to initialise it here.

Reply:
From memory, I think it is needed since the CartesianVector class does not have a default constructor